### PR TITLE
[GH-13463] add unit tests to the ChannelCreateCmd mmctl command

### PR DESCRIPTION
**Pull request by fedealconada**
#### Summary
Adds unit tests to the ChannelCreateCmd mmctl command

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13463

### CODE REVIEW

File: cmd/mattermost/commands/channel_test.go [hunk 1]
```diff
@@ -113,16 +113,58 @@ func TestRestoreChannel(t *testing.T) {
 }
 
 func TestCreateChannel(t *testing.T) {
+
```
Comment (mgdelacroix) [inline cmd/mattermost/commands/channel_test.go:116]: nit: unnecessary whitespace

Comment (fedealconada) [reply cmd/mattermost/commands/channel_test.go:116]: fixed

**Comment by fedealconada** [comment]
hey @isacikgoz, I've never seen your review, sorry about that. It seems like all tests in `cmd/mattermost/commands` do not check that the errors are the expected errors. This might be because the `err` value is always `exit status 1`. In order to check that, what I've done is to use `RunCommandWithOutput` and then check that the expected error string is a substring of the output. If there's any better way of doing that, let me know :)

File: cmd/mattermost/commands/channel_test.go [hunk 2]
```diff
@@ -117,12 +117,63 @@ func TestCreateChannel(t *testing.T) {
...
+		assert.Equal(t, name, channel.Name)
+		assert.Equal(t, model.CHANNEL_OPEN, channel.Type)
+		assert.Equal(t, "this is a header", channel.Header)
+		assert.Equal(t, "this is the purpose", channel.Purpose)
+	})
+
+	t.Run("should not create channel if name already exists on the same team", func(t *testing.T) {
+		output, err := th.RunCommandWithOutput(t, "channel", "create", "--display_name", commonName, "--team", th.BasicTeam.Name, "--name", commonName)
+		require.Error(t, err)
+		assert.True(t, strings.Contains(output, "A channel with that name already exists on the same team."))
```
Comment (isacikgoz) [inline cmd/mattermost/commands/channel_test.go:151]: I think it would be more idiomatic if we would use [require.Contains](https://pkg.go.dev/github.com/stretchr/testify/require?tab=doc#Contains). We can use this every occurrence of  `assert.True(t, strings.Contains(...`

Comment (fedealconada) [reply cmd/mattermost/commands/channel_test.go:151]: Oh, cool! Didn't know about that one, I'm new to Go hehe

Comment (fedealconada) [reply cmd/mattermost/commands/channel_test.go]: done @isacikgoz

**Comment by teresa-novoa** [comment]
@isacikgoz @mgdelacroix where are we with this PR? Thanks!

**Comment by isacikgoz** [comment]
Thanks for your contribution @fedealconada!! 🎉
